### PR TITLE
Replace EXIF Profile API

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
This PR integrates the updates made by @dlemstra to the EXIF metadata API fixing #1049 

The updates are a modified version of the API contained in [Magick.NET](https://github.com/dlemstra/Magick.NET/tree/master/src/Magick.NET/Shared/Profiles/Exif) with some further tweaks to workaround issues discovered during testing.

A big win is type safety. The API is now far more intuitive and requires much less in-depth knowledge of the EXIF specification to use.

### CODE REVIEW

File: tests/ImageSharp.Tests/Metadata/Profiles/Exif/ExifProfileTests.cs [hunk 1]
```diff
@@ -143,87 +149,94 @@ public void ReadWriteInfinity(TestImageWriteFormat imageFormat)
             image.Metadata.ExifProfile.SetValue(ExifTag.FlashEnergy, new Rational(double.NegativeInfinity));
 
             image = WriteAndRead(image, imageFormat);
-            value = image.Metadata.ExifProfile.GetValue(ExifTag.FlashEnergy);
+            IExifValue<Rational> value2 = image.Metadata.ExifProfile.GetValue(ExifTag.FlashEnergy);
             Assert.NotNull(value);
```
Comment (dlemstra) [inline tests/ImageSharp.Tests/Metadata/Profiles/Exif/ExifProfileTests.cs]: This should be `value2` instead.

File: src/ImageSharp/Metadata/Profiles/Exif/Values/ExifValues.cs [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+                case ExifTagValue.RecommendedExposureIndex: return new ExifLong(ExifTag.RecommendedExposureIndex);
+                case ExifTagValue.ISOSpeed: return new ExifLong(ExifTag.ISOSpeed);
+                case ExifTagValue.ISOSpeedLatitudeyyy: return new ExifLong(ExifTag.ISOSpeedLatitudeyyy);
+                case ExifTagValue.ISOSpeedLatitudezzz: return new ExifLong(ExifTag.ISOSpeedLatitudezzz);
+                case ExifTagValue.FaxRecvParams: return new ExifLong(ExifTag.FaxRecvParams);
+                case ExifTagValue.FaxRecvTime: return new ExifLong(ExifTag.FaxRecvTime);
+
+                case ExifTagValue.FreeOffsets: return new ExifLongArray(ExifTag.FreeOffsets);
+                case ExifTagValue.FreeByteCounts: return new ExifLongArray(ExifTag.FreeByteCounts);
+                case ExifTagValue.ColorResponseUnit: return new ExifLongArray(ExifTag.TileOffsets);
```
[the remark this answers is not in the document]
Comment (JimBobSquarePants) [reply src/ImageSharp/Metadata/Profiles/Exif/Values/ExifValues.cs]: Crap. Yes you're right!

[the remark this answers is not in the document]
Comment (brianpopow) [reply src/ImageSharp/Metadata/Profiles/Exif/Values/ExifValues.cs]: sorry for the late response. I actually reviewed it yesterday, but missed that i needed to hit the "submit review" button :sweat:

[the remark this answers is not in the document]
Comment (JimBobSquarePants) [reply src/ImageSharp/Metadata/Profiles/Exif/Values/ExifValues.cs]: Now worries. There were so many files to review! 

I've just opened #1069 with fixes.

**Pull request merged**